web/src/js/ducks/modes/regular.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    RECEIVE as RECEIVE_OPTIONS,
+    UPDATE as UPDATE_OPTIONS,
+} from "../options";
+import { ModeState, updateMode } from "../modes";
+
+export const TOGGLE_REGULAR = "TOGGLE_REGULAR";
+export const ERROR_REGULAR = "ERROR_REGULAR";
+
+interface RegularState extends ModeState {}
+
+export const initialState: RegularState = {
+    active: true,
+};
+
+export const getMode = (modes) => {
+    const regularMode = modes.regular;
+    let mode = "regular";
+    if (regularMode.active) {
+        if (regularMode.listen_host) {
+            mode += `@${regularMode.listen_host}`;
+        }
+        if (regularMode.listen_port) {
+            mode += `:${regularMode.listen_port}`;
+        }
+        return mode;
+    }
```

Comment:
This pattern repeats in almost all modes, so maybe that's a candidate for a `addListenAddr(mode)` helper? 😃